### PR TITLE
chore!: Remove Resilience Options

### PR DESCRIPTION
**Pull request by FrankEssenberger**
Relates SAP/cloud-sdk-backlog#669.

As a first step remove the old resilience options. In a next step the timeout and circuit breaker implementations are done using the middleware.

### CODE REVIEW

File: packages/connectivity/src/scp-cf/destination/destination-accessor-failure-cases.spec.ts [hunk 1]
```diff
@@ -27,12 +27,14 @@ import {
...
 import * as jwt from '../jwt';
 import { getDestination } from './destination-accessor';
+import { circuitBreaker } from './destination-service';
 
 const { wrapJwtInHeader } = jwt;
 
 describe('Failure cases', () => {
   beforeEach(() => {
     clientCredentialsTokenCache.clear();
+    circuitBreaker?.clearCache();
```
Comment (deekshas8) [inline packages/connectivity/src/scp-cf/destination/destination-accessor-failure-cases.spec.ts:37]: [q] The default `cache` option is false, do we still need this call?

File: packages/connectivity/src/scp-cf/circuit-breaker.spec.ts [hunk 2]
```diff
@@ -18,9 +17,18 @@ const jwt = jwt123.sign(
   }
 );
 
-const attempts = circuitBreakerDefaultOptions.volumeThreshold!;
-
 describe('circuit breaker', () => {
+  afterEach(() => {
+    circuitBreaker.enable();
+    nock.cleanAll();
+    // nock.restore()
```
Comment (fwilhe) [inline packages/connectivity/src/scp-cf/circuit-breaker.spec.ts:24]: intentionally commented out?

File: packages/connectivity/src/scp-cf/circuit-breaker.spec.ts [hunk 3]
```diff
@@ -18,9 +17,18 @@ const jwt = jwt123.sign(
...
+  });
+  beforeEach(() => {
+    circuitBreaker.enable();
+    nock.cleanAll();
+    // nock.restore()
```
Comment (fwilhe) [inline packages/connectivity/src/scp-cf/circuit-breaker.spec.ts:29]: intentionally commented out?

File: packages/connectivity/src/scp-cf/circuit-breaker.spec.ts [hunk 4]
```diff
@@ -31,24 +39,29 @@ describe('circuit breaker', () => {
       .get(/.*/)
       .times(1)
       .reply(200, JSON.stringify({ URL: 'test' }));
-    nock(destinationServiceUri).get(/.*/).times(attempts).reply(400);
+    nock(destinationServiceUri).get(/.*/).times(99).reply(400);
```
Comment (fwilhe) [inline packages/connectivity/src/scp-cf/circuit-breaker.spec.ts:42]: nitpick: 99 as in 'infinitely often', right? maybe use a constant for clarity.

nitpick 2 (not new in this PR): Are there no default constants for http status codes in nodejs?

File: packages/connectivity/src/scp-cf/destination/destination-service.ts [hunk 5]
```diff
@@ -299,74 +306,49 @@ type DestinationCertificateJson = {
...
-  const { enableCircuitBreaker, timeout } = {
-    ...defaultResilienceBTPServices,
-    ...options
-  };
   const config: AxiosRequestConfig = {
     ...urlAndAgent(uri),
     proxy: false,
     method: 'get',
-    timeout,
+    timeout: 10000, // TODO: Use middleware
```
Comment (fwilhe) [inline packages/connectivity/src/scp-cf/destination/destination-service.ts:347]: todos will be addressed in another pr? maybe reference the issue where it will be done here?

**Review by fwilhe — COMMENTED** [review]
few comments

File: packages/connectivity/src/scp-cf/circuit-breaker.spec.ts (hunk 4 above)
Comment (FrankEssenberger) [reply packages/connectivity/src/scp-cf/circuit-breaker.spec.ts:42]: I am not aware of the default constants.

File: packages/connectivity/src/scp-cf/circuit-breaker.spec.ts (hunk 2 above)
Comment (FrankEssenberger) [reply packages/connectivity/src/scp-cf/circuit-breaker.spec.ts]: no just old comment

File: packages/resilience/internal.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+'use strict';
```
Comment (fwilhe) [inline packages/resilience/internal.js:1]: was this file added intentionally?

**Review by fwilhe — APPROVED** [review]
lgtm

File: packages/resilience/internal.js (hunk 6 above)
Comment (FrankEssenberger) [reply packages/resilience/internal.js:1]: yes this exposes the things under the /internal path.
